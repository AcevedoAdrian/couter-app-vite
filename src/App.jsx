// Esta funciÃ³n no se va a volver a cargar en memoria cuando se realice el render de nuevo. Ya que esta fuera del functional component que hace el rende
const sum = (a, b) => {
  const c = a + b;
  return `El resultado es ${c}`;
};

export const App = () => {
  return (
    <>
      {sum(3, 3)}
      <h1>Hello World</h1>
    </>
  );
};
