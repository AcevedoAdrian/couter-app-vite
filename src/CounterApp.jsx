import { useState } from "react";
import PropTypes from "prop-types";

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  const handleSum = (e) => {
    setCounter(counter + 1);
  };
  const handleSubtract = (e) => {
    setCounter(counter - 1);
  };

  const handleReset = (e) => {
    setCounter(value);
  };

  return (
    <>
      <h1>I am the component CounterApp </h1>
      <h2> {counter} </h2>
      <button onClick={handleSum}>+1</button>
      <button onClick={handleSubtract}>-1</button>
      <button onClick={handleReset}>Reset</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};

CounterApp.defaultProps = {
  value: 32,
};
