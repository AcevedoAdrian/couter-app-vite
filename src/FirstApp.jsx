import PropTypes from "prop-types";

export const FirstApp = ({ title, subTitle }) => {
  if (!title) {
    throw new Error("The title is null");
  }

  return (
    <>
      <h2>My name is {title}</h2>
      <p>{subTitle}</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
};

FirstApp.defaultProps = {
  name: "Acevedo Adrian",
  subTitle: "Esto es un subtitulo",
  title: "Esto es un Titulo",
};
