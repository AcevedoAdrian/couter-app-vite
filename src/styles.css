head,
body {
  background-color: #080808;
  color: #ffffff;
  font-family: Helvetica, Arial, sans-serif;
  font-size: 1.3rem;
  padding: 70px;
}

button {
  padding: 10px;
  font-size: 1.3rem;
  margin-right: 5px;
}
