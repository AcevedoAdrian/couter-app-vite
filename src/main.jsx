import React from "react";
import ReactDOM from "react-dom/client";
import { App } from "./App";
import { CounterApp } from "./CounterApp";
import { FirstApp } from "./FirstApp";
import "./styles.css";
ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    {/* <App /> */}
    {/* <FirstApp subTitle="About My" /> */}
    <CounterApp value={10} />
  </React.StrictMode>
);
