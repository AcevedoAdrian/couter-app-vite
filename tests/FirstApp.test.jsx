import { render } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe("Pruebas en FirstApp", () => {
  test("Debe de hacer match con el snapshot", () => {
    const title = "Hola Soy Adrian";
    // Esto renderiza el componente en memoria, actualiza el objeto screen que es un objeto propio de react testing librery. Y tambien retorna un objetos que expone ciertas propiedades. Una de ellas es el container, que es un objeto muy parecido al DOM
    const { container } = render(<FirstApp title={title} />);

    // Esto crea un snapshot(toma una fotografia) del componente renderizado en memoria y lo guarda en una carpeta llamada __snapshots__ con el nombre del archivo donde estoy ejecutando. Esta captura la hace graicas a que render retorna el container el cual es como un nodo de html del componente renderizado y con toMatchSnapshot guardamos ese contenido en un archivo.
    //expect(container).toMatchSnapshot();
  });

  test("debe de mostrar el titulo en un h1", () => {
    const title = "Hola Soy Adrian";
    // Obtiene un nodo con texto
    const { container, getByText, getByTestId } = render(
      <FirstApp title={title} />
    );
    expect(getByText(title)).toBeTruthy();
    // Buscando un h1 en el componente rensderizado
    // const h1 = container.querySelector("h2");
    // Esto es una hacercion
    // expect(h1.innerHTML).toBe(title);
    // Evalua que contenga dentro del elemento el contenido de title
    // expect(h1.innerHTML).toContain(title);

    // Obteniendo elementos desde el html con un data-test
    // Verifica si este elemento existe en el html
    expect(getByTestId("test-title")).toBeTruthy();

    // verifica si existe el elementento con el contenido indicado en el tobe
    expect(getByTestId("test-title").innerHTML).toBe(title);

    //
  });

  test("Debe de mostrar el subtitulo enviado por props", () => {
    const title = "Hola, Soy Adrian";
    const subTitle = "Hola, soy un subtitulo";
    const { getAllByText } = render(
      <FirstApp title={title} subTitle={subTitle} />
    );

    expect(getAllByText(subTitle).length).toBe(2);
  });
  
});
