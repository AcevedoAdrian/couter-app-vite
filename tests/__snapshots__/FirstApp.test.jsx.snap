// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp Debe de hacer match con el snapshot 1`] = `
<div>
  <h2>
    Hola Soy Adrian
  </h2>
  <p>
    Esto es un subtitulo
  </p>
</div>
`;
